[[ Bug 21769 ]] Add new function iphoneDeviceModel()

livecodepanos opened the pull request.

livecodepanos commented:
![IMG_B40E38FCE543-1](https://user-images.githubusercontent.com/5111835/96723346-2378e300-13b7-11eb-90a4-b2a5b6ccd648.jpeg)

This is what `iphoneDeviceModel()` returns in my iPhone7 device

livecodepanos commented on engine/src/mbliphoneapp.mm:
```diff
@@ -1717,63 +1717,140 @@ - (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
+									   @"iPhone 6S",            @"iPhone8,1",
+									   @"iPhone 6S Plus",       @"iPhone8,2",
+									   @"iPhone SE",            @"iPhone8,4",
+									   @"iPhone 7 (CDMA)",      @"iPhone9,1",
```
@runrevmark I am wondering, do we need to differentiate between e.g. `"iPhone 7 (CDMA)"`  and `"iPhone 7 (GSM)"`, or just a `iPhone 7` is enough for both cases? I guess the developer only cares if the device is an iPhone 7, not if it is GSM or CDMA.

runrevmark replied:
I'm not sure we should be mapping the model as reported by the system at all - i.e. just return things like iPhone11,2

runrevmark reviewed (changes requested):
So I think the mapping to 'english' names should be removed as it doesn't add anything and only makes it more difficult to use (due to variation in how names are structured).

There is another internal caller to the core function here - but I think that can be removed rather than updated to use the standard naming since I think its checking for models we no longer support?

livecodepanos committed:
[[ Bug 21769 ]] Added new function iphoneDeviceModel()

This function returns the full name of the device the app is running on.
If the app is running on a simulator, it returns "i386" or "x86_64".

livecodeian commented on docs/notes/bugfix-21769.md:
```diff
@@ -0,0 +1 @@
+# Added new function iphoneDeviceModel() that returns the full name of the iOS device the app is running on
```
@livecodepanos This is inaccurate as the function no longer maps the machine name to a readable full device name

livecodeian commented:
@livecodepanos This needs a dictionary entry for iphoneDeviceModel

livecodepanos committed:
Merge remote-tracking branch 'upstream/develop-9.6' into iphone_get_device_model

livecodepanos committed:
[[ Bug 21769 ]] Added documentation for iphoneDeviceModel()

This patch adds a dictionary entry and corrects the release note for
iphoneDeviceModel() function

livecodeian commented:
@livecode-vulcan review ok 85f29c7

livecode-vulcan commented:
💙 review by @livecodeian ok 85f29c7cf381f1e186cbbb93c9b718028bca2d37

livecode-vulcan commented:
😎 test success 85f29c7cf381f1e186cbbb93c9b718028bca2d37

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1029)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/437)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/436)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/435)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/436)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2554)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/31)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/83)
* try-community-universal-ios-iphonesimulator14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/31)
* try-community-universal-ios-iphonesimulator14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/82)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2580)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2570)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2568)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2605)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2541)
